### PR TITLE
Laboratory work #2, Varvara Migacheva - 22FPL2

### PR DESCRIPTION
**Pull request by varmig2000**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Very good. What about 6, 8, 10?

**Comment by varmig2000** [comment]
I tried...... Sorry. See the first "step 2"

File: lab_2_tokenize_by_bpe/main.py [hunk 1]
```diff
@@ -14,6 +14,21 @@ def prepare_word(
...
     :return: preprocessed word
     """
+    if not (
+            isinstance(raw_word, str) and
+            (isinstance(start_of_word, str) or start_of_word is None) and
+            (isinstance(end_of_word, str) or end_of_word is None)
+    ):
+        return None
+    tokens = []
+    for el in raw_word:
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:24]: simplify

File: lab_2_tokenize_by_bpe/main.py [hunk 2]
```diff
@@ -47,6 +88,22 @@ def merge_tokens(
...
+    ):
+        return None
+    merged_dictionary = {}
+    for el in word_frequencies:
+        el_list = list(el)
+        for i in range(len(el) - 1):
+            tokens = (el[i], el[i + 1])
+            if tokens == pair:
+                el_list.pop(i + 1)
+                el_list[i] = pair[0] + pair[1]
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:103]: join

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**